### PR TITLE
UNet2D HipMRI Segmentation (47443349)

### PR DESCRIPTION
**Pull request by AlexPitman04**
Name: Alexander Pitman
Student ID: 47443349

This pull request includes my implementation of a UNet2D, trained to segment processed medical image slices from the HipMRI Study on Prostate Cancer according to Task 1 on the COMP3710 Project Task Sheet.

The trained UNet2D model achieved an average Dice Score of 0.985, 0.981, 0.922, 0.956, 0.758, and 0.702 on the test set for the Background, Body, Bone, Bladder, Rectum, and Prostate labels respectively.

I have structured my project as follows:

README.MD: contains project documentation including a description the dataset, a description of my model, how it works, the problem it solves, the results, dependencies, and instructions for usage.

dataset.py: contains data pipeline functionality.

modules.py: contains UNet2D network architecture.

train.py: contains model training algorithm.

predict.py: contains model testing where predictions are made on the test set with dice scores reported and sample predictions visualised.

utils.py: contains various global constants and utility functions such as loss functions, dice score calculation, and seed setting.

images/: contains all the images used in the README.

### CODE REVIEW

**Comment by yexincheng** [comment]
**This is an initial inspection, no action is required at this point**
1. **Solves problem**: The problem was well solved, 5 labels reached over 0.75 Dice score and one label reached 0.7 Dice score. Training loss, prediction figures and metrics were provided. Nice and well-structured documentation. 
2. **Implementation**: Nice uni tests. Data augmentation implemented. Some hpyerparameters were hard-coded. Good and clear implementation
3. **Design**: Great
4. **Comment**: Sufficient
5. **Difficulty**: Easy

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>Header blocks. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>Good solution to problem. </td><td></td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting present. </td><td></td></tr>
	<tr><td></td><td>No Data leakage found. </td><td></td></tr>
	<tr><td></td><td>Difficulty : Easy. UNet2D (Easy Difficulty)</td><td>-10</td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Good Meaningful commit messages. </td><td></td></tr>
	<tr><td></td><td>Good Progressive commits. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Good. </td><td></td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Good. </td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). Submitted on time.</td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is good. </td><td></td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-10</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by shakes76** [comment]
Segmentations not one hot encoded -2 additionally
